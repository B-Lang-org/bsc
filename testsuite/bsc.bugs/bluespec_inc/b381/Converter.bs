-- #############################################################################
-- #
-- #############################################################################

package Converter(Converter(..), mkConverter) where

-- #############################################################################
-- #
-- #############################################################################

interface Converter i =
    convert :: i -> (i, i)

interface VConverter ni =
    convert :: Bit ni -> Bit mi

vMkConverter :: Module (VConverter i)
vMkConverter =
    module verilog "Converter" (("foo", (valueOf i))) "CLK" {
	convert = "IN" "OUT";
    } [ convert <> convert ]

mkConverter :: (IsModule m c, Bits i si) => m (Converter i)
mkConverter = liftModule $
     module
      _a :: VConverter si
      _a <- vMkConverter
      interface
	convert i = unpack (_a.convert (pack i))

-- #############################################################################
-- #
-- #############################################################################

