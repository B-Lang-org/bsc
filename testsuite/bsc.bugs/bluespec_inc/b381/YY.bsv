////////////////////////////////////////////////////////////////////////////////
///
////////////////////////////////////////////////////////////////////////////////

package YY;

////////////////////////////////////////////////////////////////////////////////
///
////////////////////////////////////////////////////////////////////////////////

import Converter::*;

////////////////////////////////////////////////////////////////////////////////
///
////////////////////////////////////////////////////////////////////////////////

(* synthesize *)
module mkYY (Empty);

   Converter#(Bit#(8))       y();
   mkConverter               y_reg(y);

endmodule

endpackage

////////////////////////////////////////////////////////////////////////////////
///
////////////////////////////////////////////////////////////////////////////////

