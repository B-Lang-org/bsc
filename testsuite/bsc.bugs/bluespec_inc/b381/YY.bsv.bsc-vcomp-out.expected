checking package dependencies
compiling YY.bsv
code generation for mkYY starts
Internal Bluespec Compiler Error:
Please report this failure to your Bluespec technical contact.
If you do not know your contact, you may email support@bluespec.com.
The following internal error message should be included in your
correspondence along with any other relevant details:
Bluespec Compiler (build 5c1b4dc)
aTypeConv|ITForAll _v101[IdP_bad_name] IKNum (ITAp (ITAp (ITCon Prelude::-> (IKFun IKStar (IKFun IKStar IKStar)) TIabstract) (ITAp (ITCon Prelude::Bit (IKFun IKNum IKStar) TIabstract) (ITNum 8))) (ITAp (ITCon Prelude::Bit (IKFun IKNum IKStar) TIabstract) (ITVar _v101[IdP_bad_name])))
